### PR TITLE
Set internal gloabls after sourcing input config file

### PR DESCRIPTION
**Pull request by rhvgoyal**
This will make sure that config file can't set any global
variables of container-storage-setup which are internal and
change behavior unexpectedly.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

### CODE REVIEW

**Comment by rhvgoyal** [comment]
@rhatdan PTAL

File: container-storage-setup.sh [hunk 1]
```diff
@@ -21,40 +21,42 @@
...
+  # 'man container-storage-setup'
 
-DOCKER_ROOT_LV_NAME="docker-root-lv"
-DOCKER_ROOT_DIR="/var/lib/docker"
-DOCKER_METADATA_DIR="/var/lib/docker"
-DOCKER_ROOT_VOLUME_SIZE=40%FREE
+  DOCKER_ROOT_LV_NAME="docker-root-lv"
+  DOCKER_ROOT_DIR="/var/lib/docker"
+  DOCKER_METADATA_DIR="/var/lib/docker"
+  DOCKER_ROOT_VOLUME_SIZE=40%FREE
```
Comment (rhatdan) [inline container-storage-setup.sh]: Can't these be overriden by the user?
@shishir-a412ed PTAL

File: container-storage-setup.sh [hunk 2]
```diff
@@ -21,40 +21,42 @@
...
 
-STORAGE_IN_FILE="/etc/sysconfig/docker-storage-setup"
-STORAGE_OUT_FILE="/etc/sysconfig/docker-storage"
-STORAGE_DRIVERS="devicemapper overlay overlay2"
+  STORAGE_IN_FILE="/etc/sysconfig/docker-storage-setup"
```
Comment (rhatdan) [inline container-storage-setup.sh]: These are set above and need to be set during getopt?

File: container-storage-setup.sh [hunk 3]
```diff
@@ -1199,8 +1201,34 @@ elif [ -e /usr/share/container-storage-setup/container-storage-setup ]; then
...
+# that file.
+if [ $# -eq 2 ]; then
+  [ -e "$1" ] && source "$1"
+else
+  if [ -e "/etc/sysconfig/docker-storage-setup" ]; then
+    source /etc/sysconfig/docker-storage-setup
+  fi
+fi
+
+set_internal_gloabls
```
Comment (rhatdan) [inline container-storage-setup.sh]: typo?

Comment (rhvgoyal) [inline container-storage-setup.sh]: yes

File: container-storage-setup.sh (hunk 1 above)
Comment (rhvgoyal) [inline container-storage-setup.sh]: DOCKER_ROOT_VOLUME_SIZE can be. I think I should divide them in two parts. One which can be overridden and once which are internal. In fact, I should probably prefix internal ones with CSS_FOO.

File: container-storage-setup.sh (hunk 2 above)
Comment (rhvgoyal) [inline container-storage-setup.sh]: Not sure what you are referring to.

Comment (rhatdan) [inline container-storage-setup.sh]: I looked at your code again, and it is ok.  I thought you were setting this after you set
STORAGE_IN_FILE=$1
STORAGE_OUT_FILE=$2

File: container-storage-setup.sh (hunk 1 above)
Comment (rhatdan) [inline container-storage-setup.sh]: Yes this is why I was adding the _ to indicate internal (In Python anyways.)

File: container-storage-setup.sh (hunk 2 above)
Comment (rhvgoyal) [inline container-storage-setup.sh]: I am doing getopt twice now. internal gloabls are set after first getopt. That getopt is done just to figure out which config file should be sourced.

I source the file and after that I do getopt again and that time override STORAGE_IN_FILE and STORAGE_OUT_FILE

**Comment by rhvgoyal** [comment]
This is not needed any more after merge of PR #220

**Pull request closed without merging**